Automattic/wp-calypso
# [WordAds] Update ad dashboard for free sites removed from wordads

## Proposed Changes

Going forward sites enrolled in WordAds would be temporarily removed once they no longer have a valid WordPress.com plan. This PR updates the UX for those sites to:
- Inform them they would be removed from WordAds in the process of canceling their plan.
- Allow them to view their earning and payment history on the Earnings dashboard but prevent updates to Settings
- Provide them an option to rejoin WordAds by upgrading.

## Testing Instructions

* Create a new WordPress.com site and apply to join WordAds under, `My site -> Tools -> Earn -> Earn ad revenue`
* You would subscribe to a premium plan or higher (as part of Step 1)
* Navigate to `My site -> Upgrades -> Manage plan -> Cancel plan`, by clicking on the cancel plan button you should see a modal informing you of the consequences of canceling a plan.

 _In a scenario where the site has a primary domain the message will be combined_
<img width="1728" alt="notice" src="https://user-images.githubusercontent.com/9039613/219135431-b457e183-e96a-4d79-9a2c-151671a98fd2.png">

_In case the site has no primary domain at the time the message will be independent_
<img width="1728" alt="no primary domain" src="https://user-images.githubusercontent.com/9039613/219447859-c7fbbf12-3b30-4303-a3d9-d60348389d00.png">

* Now we need to simulate a subscription plan expiry by manually removing the subscription via the StoreAdmin. Before removing the subscription, make sure you are running the D100709-code revision in your sandbox
* Once you remove the subscription your WordAds site should now have an ineligible status
* Navigate to `My site  -> Tools -> Earn -> View ad dashboard`
<img width="1726" alt="1" src="https://user-images.githubusercontent.com/9039613/219136885-b3256f08-e8cb-4323-b6b3-df62430adf41.png">

* You should be able to view your Earnings and payment history, but the Settings tab should only have a preview with options disabled
*  You should also see an upgrade banner on top of the page
<img width="1720" alt="settings" src="https://user-images.githubusercontent.com/9039613/219137032-f0ef1161-6329-4820-8865-64f8b14a35a9.png">

* Upgrade your plan by clicking the "Upgrade" button
* After upgrading, navigate back to `My site -> Tools -> Earn -> View ad dashboard`, you should no longer see the upgrade banner and you should be able to make changes to the site Settings


## Pre-merge Checklist
